prow/git/git.go

```diff
@@ -295,6 +295,11 @@ func (r *Repo) Merge(commitlike string) (bool, error) {
 	return r.MergeWithStrategy(commitlike, prowgithub.MergeMerge)
 }
 
+// MergeWithMessage is the same with Merge() but with a message included.
+func (r *Repo) MergeWithMessage(commitlike, message string) (bool, error) {
+	return r.mergeWithMergeStrategyMergeAndMessage(commitlike, message)
+}
+
 // MergeWithStrategy attempts to merge commitlike into the current branch given the merge strategy.
 // It returns true if the merge completes. It returns an error if the abort fails.
 func (r *Repo) MergeWithStrategy(commitlike string, mergeStrategy prowgithub.PullRequestMergeType) (bool, error) {
```

Comment:
@droslean can we solve this via functional opts pattern?

```
// unexported, we don;t want to allow everything
type extraArg []string

type MergeOpt func() extrArg
func WithMessage(message string) MergeOpt {
  return func() extraArg {
    return []striing{"-m", message"}
  }
}

func (r *Repo) MergeWithStrategy(commitlike string, mergeStrategy prowgithub.PullRequestMergeType, opts ...MergeOpt)
```

Otherwise the number of funcs we need will explode because we need one for each git operation + possible extra arg combination